src/source/linear_ramp.rs

```diff
@@ -0,0 +1,124 @@
+use std::time::Duration;
+
+use crate::{Sample, Source};
+
+/// Internal function that builds a `LinearRamp` object.
+pub fn linear_gain_ramp<I>(
+    input: I,
+    duration: Duration,
+    start_gain: f32,
+    end_gain: f32,
+) -> LinearGainRamp<I>
+where
+    I: Source,
+    I::Item: Sample,
+{
+    let duration = duration.as_secs() * 1000000000 + duration.subsec_nanos() as u64;
+
+    LinearGainRamp {
+        input,
+        remaining_ns: duration as f32,
```

Comment:
Should we have a f32::min(duration, 1.0) here? That would prevent a div by zero if the user puts in a duration of zero. (in lerp). On the other hand maybe the user wants to know when they make that mistake. Maybe an assert with a good message?